internal/parser/parser.go

```diff
@@ -94,6 +103,7 @@ func getParser() *Parser {
 
 func putParser(p *Parser) {
 	*p = Parser{scanner: p.scanner}
+	p.initializeClosures()
```

Comment:
```suggestion
	*p = Parser{scanner: p.scanner, setParentFromContext: p.setParentFromContext}
```

Otherwise this will just reclose each time the parser is put